posts/2021-01-06-ocp-log-forwarding.adoc

```diff
@@ -0,0 +1,424 @@
++
+*Note:* The cluster logging instance sample that is provided in the OpenShift documentation leaves the _storage class_ value for the reader to configure. In test environments, you can omit the storage section and the cluster logging instance uses in-memory storage. However, this configuration is not suitable for production.
+
+[start=2]
+. [[keycert-secret]]*Optional:* It is recommended that you secure the connection between the Fluentd servers on your OpenShift cluster and the external Fluentd server. This step creates a _secret_ that is used by the Log Forwarding API to achieve a secure connection. This procedure assumes that the necessary key and certificate files, for example `tls.crt` and `tls.key`, are already created. The instructions for creating the keys and certificates are not within the scope of this blog post. The key and certificate are also used when you <<keycert-fluentd, deploy the external Fluentd server>>.
++
+.. Switch to the `openshift-logging` project
++
+[source]
+----
+oc project openshift-logging
+----
+.. Create a secret that contains your key and certificate. You also need to specify a _shared_key_, which is a string password that is used as an extra layer of authentication between the Fluentd servers on the OpenShift cluster and the external Fluentd server. The examples in this blog post use _secure-forward_ for the secret name and  _"secretpassword"_ for the shared key. Change the paths in the following example as necessary:
++
+[source]
+----
+oc create secret generic secure-forward --from-file=ca-bundle.crt=/path/to/tls.crt --from-file=tls.crt=/path/to/tls.crt --from-file=tls.key=/path/to/tls.key   --from-literal=shared_key=secretpassword
+----
+
+
+== Configuring the Log Forwarding API
+
+
+. Create the `log-forward-instance.yaml` file. Configure the outputs and pipelines.
++
+* The `outputs` section of the file specifies where to send the logs. To establish a secure connection to an output, you must specify a secret, such as the one that you created in <<keycert-secret,step 2b of the *Before you begin* section>>. If you want to use an insecure connection for test purposes, you can omit the secret from your output configuration.
+* The `pipelines` section of the file specifies which log types are sent to the configured outputs.
++
+The following sample `log-forward-instance.yaml` file defines two `outputs`: a secure connection to a Fluentd server named `fluentd-server-secure` and an insecure connection to another Fluentd server named `fluentd-server-insecure`. Note the different protocols that are defined for the _url_ in each output (tls vs tcp). Remove the output type that you do not want to use in your file. Each log type and its corresponding output reference is defined in the `pipelines` section. The `default` output reference points to the internal OpenShift Container Platform Elasticsearch instance:
++
+```
```

Comment:
should one of the pipelines (`infra-logs` or `audit-logs`) reference the `fluentd-server-insecure` output? We've defined that output in the spec but not referenced it in an actual pipeline. The last two pipelines only reference the `default` output, which isn't really the focus here- should one of them also reference `fluentd-server-insecure` for demonstrative value?